server/src/cg/group4/server/database/query/GetGroupData.java

```diff
@@ -14,22 +14,24 @@
 public class GetGroupData extends Query {
 
     @Override
-    public ArrayList<GroupData> query(DatabaseConnection databaseConnection) throws SQLException {
+    public ArrayList<GroupData> query(final Connection databaseConnection) throws SQLException {
         ArrayList<GroupData> list = new ArrayList<GroupData>();
-        Statement statement = databaseConnection.query();
-        ResultSet resultSet = statement.executeQuery(
-                "SELECT G.Key AS GroupId, G.Name AS Name, G.OwnerId AS OwnerId, U.Username AS Username, U.Id "
-                        + "FROM 'Group' G INNER JOIN User U ON G.OwnerId = U.Id");
-        while (resultSet.next()) {
-            list.add(new GroupData(
-                    resultSet.getInt("GroupId"),
-                    resultSet.getString("Name"),
-                    resultSet.getString("OwnerId"),
-                    resultSet.getString("Username")
-            ));
+
+        String query = "SELECT G.Key AS GroupId, G.Name AS Name, G.OwnerId AS OwnerId, U.Username AS Username, U.Id "
+                + "FROM 'Group' G INNER JOIN User U ON G.OwnerId = U.Id";
```

Comment:
Same here... Remote server should not contain user data... and local database should not contain group data...